deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/FileSplitDataSetIterator.java

```diff
@@ -0,0 +1,127 @@
+package org.deeplearning4j.datasets.iterator;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.datasets.iterator.callbacks.FileCallback;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.dataset.api.DataSetPreProcessor;
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator;
+import org.nd4j.linalg.factory.Nd4j;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Simple iterator working with list of files.
+ * File -> DataSet conversion will be handled via provided FileCallback implementation
+ *
+ * @author raver119@gmail.com
+ */
+@Slf4j
+public class FileSplitDataSetIterator implements DataSetIterator {
+    private DataSetPreProcessor preProcessor;
+
+    private List<File> files;
+    private int numFiles;
+    private AtomicInteger counter = new AtomicInteger(0);
+    private FileCallback callback;
+
+    public FileSplitDataSetIterator(@NonNull List<File> files, @NonNull FileCallback callback) {
+        this.files = files;
+        this.numFiles = files.size();
+        this.callback = callback;
+    }
+
+
+    @Override
+    public DataSet next(int num) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int totalExamples() {
+        return numFiles;
```

Comment:
+1 to DataSetIterator change. It should be just Iterator copy, like MultiDataSetIterator